collections/index.js

```diff
@@ -20,46 +20,126 @@ import employees from './data.json';
 }
 
 /*
   I expect this function to return me an array of employees depending on which
   department is passed in
 */
 export const getEmployeesByDepartment = (dept) => {
-
+  // sets an empty array ready to accept any values we add later:
+  const departmentEmployees = [];
+  // using a for loop to index through each of the employees on record
+  for (let i = 0; i < employees.length; i++) {
+    // in each of the iterations, if the employee 'department' matches the
+    // provided arguement 'dept', the statement is true...
+    if (employees[i].department === dept) {
+      // this adds (pushes) the employees details to the departmentEmployees
+      // array we created earlier.
+      departmentEmployees.push(employees[i]);
+    }
+  }
+  // returns all employees that match the department or an empty array if
+  // there were no matches.
+  return departmentEmployees;
+  // side note: i wrote this completely before testing and it worked first time!
 }
 
 /*
   I expect this function to look through all the employees and return an array
   of employees who are missing any data
 */
-export const getEmployeesWithMissingData = () => {
 
+// export const getEmployeesWithMissingData1 = () => {
+//   // sets an empty array ready to accept any values we add later:
+//   const missingInfoEmployees = [];
+//   // using a for loop to index through each of the employees on record
+//   for (let i = 0; i < employees.length; i++) {
+//     // checks if any of the values are empty. I prefer this method over others
+//     // I thought of because if more than one value is empty per employee, this
+//     // function only adds the employee to the array once.
+//     if (employees[i].id === "" || employees[i].firstName === "" || employees[i].lastName === "" || employees[i].department === "" || employees[i].contactNumber === "" || employees[i].taxCode === "" || employees[i].salary === "") {
+//       // this adds (pushes) the employees details to the missingInfoEmployees
+//       // array we created earlier.
+//       missingInfoEmployees.push(employees[i]);
+//     }
+//   }
+//   // returns all employees that are missing data
+//   return missingInfoEmployees;
+// }
+
+// I thought of a 2nd way of doing this which is probably better but might
+// need some tidying up.
+
+export const getEmployeesWithMissingData = () => {
+  // sets an empty array ready to accept any values we add later:
+  const missingInfoEmployees = [];
+  // using a for loop to index through each of the employees on record
+  for (let i = 0; i < employees.length; i++) {
+    // creates an array of the employee details
+    let employeeDetails = Object.values(employees[i])
+    // checks each value in the new array for any empty strings or undefined
+    employeeDetails.find(function(item) {
+      if (item === "" || item === undefined) {
+        // adds the employee to the missingInfoEmployees array
+        missingInfoEmployees.push(employees[i]);
+      }
+    })
+  }
```

Comment:
You're using `.find()` like a for-loop here - theres a function to check if an array contains a simple value, check your array test :)